File: backEnd/models/user_model.js

const mongoose = require('mongoose')


const userSchema = new mongoose.Schema({
    fullname : {
        type: String, 
        required:true,},

    imageURL : {
        type:String,
        default:'default.jpg'},

    email: {
        type:String,
        required:true,
        unique:true,},

    phone_no:  {
        type:String,
        required:true},
    password : {
        type:String, 
        required:true,},
        
    role: {
        type:String,
         default:"User"}
    });

const User = mongoose.model('User', userSchema)
module.exports = User;

File: front-end/lib/infrastructure/local_sqlite/models/user_model.dart
class User {
  final int fullname;
  final String imageURL;
  final String email;
  final String phone_no;
  final int password;
  final int role;
  final int location;

  User({ required this.fullname, required this.imageURL, required this.email, required this.phone_no, required this.password, required this.role, required this.location });

  Map<String, dynamic> toMap() {
    return {
      'fullname': fullname,
      'imageURL': imageURL,
      'email': email,
      'phone_no': phone_no,
      'password': password,
      'role': role,
      'location': location,
    };
  }

  @override
  String toString() {
    return 'Trans{fullname: $fullname, imageURL: $imageURL, email: $email, role: $role, location: $location}';
  }

}